fe/fe-core/src/main/java/com/starrocks/sql/optimizer/PlanContext.java

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.starrocks.sql.optimizer;
+
+import com.starrocks.sql.optimizer.base.ColumnRefFactory;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+
+import java.util.List;
+
+public class PlanContext {
+    private final OptExpression plan;
+    private final List<ColumnRefOperator> outputColumns;
+    private final ColumnRefFactory refFactory;
```

Comment:
move to optimizerContext, and only transmit `plan` to mv？

Reply:
As it's only used in `MaterializedView `, move it into `MaterializedView `as a inner class?

Reply:
it is only used by Materialized view

Reply:
mv PlanContext into MaterializedView as inner class